pkg/kubectl/cmd/log.go

```diff
@@ -111,13 +117,19 @@ func RunLog(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 	}
 
 	var container string
-	if len(args) == 1 {
-		if len(pod.Spec.Containers) != 1 {
-			return fmt.Errorf("POD %s has more than one container; please specify the container to print logs for", pod.ObjectMeta.Name)
-		}
-		container = pod.Spec.Containers[0].Name
+	if cmdutil.GetFlagString(cmd, "container") != "" {
+		// [-c CONTAINER]
+		container = p.containerName
 	} else {
-		container = args[1]
+		// [CONTAINER] (container as arg not flag) is supported as legacy behavior
```

Comment:
Put this PR number in the comment? Always pays to be kind to the poor soul who reads this comment 4 months from now and wonders if it can be removed yet (it might be you).